test/integration/targets/apt/tasks/main.yml

```diff
@@ -15,69 +15,99 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
- - include: 'apt.yml'
-   when: ansible_distribution in ('Ubuntu', 'Debian')
-
- - block:
-     - include: 'repo.yml'
-   always:
-     - apt_repository:
-         repo: "deb file:{{ repodir }} ./"
-         state: absent
-     - file:
-         name: "{{ repodir }}"
-         state: absent
-   when: ansible_distribution in ('Ubuntu', 'Debian')
-
- - include: 'apt-multiarch.yml'
-   when: ansible_distribution in ('Ubuntu', 'Debian') and ansible_userspace_architecture != apt_foreign_arch
-
- - include: 'apt-builddep.yml'
-   when: ansible_distribution in ('Ubuntu', 'Debian')
-
- - include: upgrade.yml upgrade_type=dist
-   when: ansible_distribution in ('Ubuntu', 'Debian')
-
- - name: Check if aptitude is installed
-   command: dpkg-query --show --showformat='${db:Status-Abbrev}' aptitude
-   register: aptitude_status
-   when: ansible_distribution in ('Ubuntu', 'Debian')
-
- - debug: var=aptitude_status.stdout
-
- - name: Remove aptitude, if installed, to test fall-back to apt-get
-   apt:
-     pkg: aptitude
-     state: absent
-   when:
-     - aptitude_status.stdout.find('ii') != -1
-
- - include: "upgrade.yml aptitude_present={{ False | bool }} upgrade_type={{ item.upgrade_type }} force_apt_get={{ item.force_apt_get }}"
-   when:
-     - ansible_distribution in ('Ubuntu', 'Debian')
-   with_items:
-     - { upgrade_type: safe, force_apt_get: False }
-     - { upgrade_type: full, force_apt_get: False }
-     - { upgrade_type: safe, force_apt_get: True }
-     - { upgrade_type: full, force_apt_get: True }
-
- - name: (Re-)Install aptitude, run same tests again
-   apt:
-     pkg: aptitude
-     state: present
-
- - include: upgrade.yml upgrade_type={{ item.upgrade_type }} force_apt_get={{ item.force_apt_get }}
-   when:
-     - ansible_distribution in ('Ubuntu', 'Debian')
-   with_items:
-     - { upgrade_type: safe, force_apt_get: False }
-     - { upgrade_type: full, force_apt_get: False }
-     - { upgrade_type: safe, force_apt_get: True }
-     - { upgrade_type: full, force_apt_get: True }
-
- - name: Remove aptitude if not originally present
-   apt:
-     pkg: aptitude
-     state: absent
-   when:
-     - aptitude_status.stdout.find('ii') == -1
+- block:
+  - name: Remove mysql which causes the upgrade tests to fail
+    block:
+      - service:
+          name: mysql
+          state: stopped
+
+      - apt:
+          name: mysql*
+          state: absent
+          purge: yes
+          autoremove: yes
+          update_cache: yes
+
+      - file:
+          name: "{{ item }}"
+          state: absent
+        with_items:
+         - /etc/mysql
+         - /var/log/mysql
+         - /var/lib/mysql
+    when: ansible_distribution_version is version('16.04', '==')
+
+  # https://github.com/Microsoft/WSL/issues/143#issuecomment-209072634
```

Comment:
We shouldn't be making system-wide changes like this, especially for WSL. I'd be OK with having a comment here explaining how someone using WSL can work around the issue though.

Reply:
I don't think it's related to WSL, if I understood it correctly, the script fixes a packaging issue in the udev package which prevents the package from being upgraded.

Reply:
To be clear, the failure the script fixes occurs in our docker env too, it just happened that I found the "fix" on WSL's github repo.